File: app/src/main/java/com/afollestad/photoaffix/animation/ViewHideAnimationListener.java
package com.afollestad.photoaffix.animation;

import android.animation.Animator;
import android.view.View;

/** @author Aidan Follestad (afollestad) */
public class ViewHideAnimationListener implements Animator.AnimatorListener {

  private final View view;

  public ViewHideAnimationListener(View view) {
    this.view = view;
  }

  @Override
  public void onAnimationStart(Animator animation) {}

  @Override
  public void onAnimationEnd(Animator animation) {
    view.setVisibility(View.GONE);
  }

  @Override
  public void onAnimationCancel(Animator animation) {}

  @Override
  public void onAnimationRepeat(Animator animation) {}
}

File: app/src/main/java/com/afollestad/photoaffix/animation/ViewHideAnimationListener.kt
/*
 * Licensed under Apache-2.0
 *
 * Designed and developed by Aidan Follestad (@afollestad)
 */
package com.afollestad.photoaffix.animation

import android.animation.Animator
import android.view.View
import android.view.View.GONE

/** @author Aidan Follestad (afollestad) */
class ViewHideAnimationListener(private val view: View) : Animator.AnimatorListener {

  override fun onAnimationStart(animation: Animator) = Unit

  override fun onAnimationEnd(animation: Animator) {
    view.visibility = GONE
  }

  override fun onAnimationCancel(animation: Animator) = Unit

  override fun onAnimationRepeat(animation: Animator) = Unit
}
